### PR TITLE
increase priority of PTF repo on all nodes

### PR DESCRIPTION
**Pull request by aspiers**
I can't think why anyone would ever place a package in the PTF repo and then *not* want it to be installed in preference to a different version from another repo.  So it makes sense for the PTF repo to have a higher priority than all other repos.

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2032,12 +2032,11 @@ function onadmin_proposal()
     if iscloudver 5plus; then
         update_one_proposal dns default
     fi
-    if [ "$networkingplugin" = "vmware" ] && iscloudver 5plus ; then
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: @aspiers could you please add to the commit message, why you removed this check? it just says it should always happen

**Comment by mjura** [comment]
Why we would like to change priority for PTF repository? I did it only for VMWare NSX testing, because for SUSE Cloud 5 we are shipping openvswitch packages with VMWare patches in lower version than openvswitch packages on our official SUSE Cloud 5 ISO.

**Comment by mjura** [comment]
In my opinion it shouldn't be general rule.

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: @aspiers just saw your comment in https://github.com/SUSE-Cloud/automation/commit/5b2827557ba42222d77296744ee34529d33480b8
wasn't referenced here in this PR

Comment (aspiers) [inline scripts/qa_crowbarsetup.sh]: @MaximilianMeister The commit message already explained why.  Did you read the last sentence?

**Comment by aspiers** [comment]
@mjura Why would anyone place a package in the PTF directory unless they wanted it to be used?

**Comment by bmwiedemann** [comment]
We could also set the prio via barclamp-provisioner's autoyast.xml.erb - because normally this is empty and if you put files in there, you normally want them used.

**Comment by aspiers** [comment]
> We could also set the prio via barclamp-provisioner's autoyast.xml.erb - because normally this repo is empty and if you put files in there, you want them used.

Yes, that's a good idea, although it's not sufficient: it does not help with the admin node or nodes installed via `crowbar_register`.

**Comment by mjura** [comment]
@aspiers For VMWare NSX integration we are shipping separate packages which contain proprietary software.

**Comment by mjura** [comment]
Anyway if you want change priority for PTF repository, I'm open.

**Comment by aspiers** [comment]
@mjura You said "In my opinion it shouldn't be general rule" so I was interested to know why you thought that.  I think it *should* be the general rule, because I can't think why anyone would ever place a package in the PTF repo and then *not* want it to be installed in preference to a different version from another repo.  But thanks for your last comment  - good to know you are not giving a -1.

**Comment by dirkmueller** [comment]
IMHO the ptf repository should not be changed generally. We want to respect packet versions and  release numbers, and if there is outdated crap in the PTF channel it shouldn't get installed.. -1

**Comment by aspiers** [comment]
> IMHO the ptf repository should not be changed generally

I agree, but I am not proposing that, only that it should be changed in `mkcloud`.

There is no programmatic way to tell whether older packages in the PTF channel are "crap" or not.  As the commit messages already explain, they will often *not* be crap because PTF packages built from older PR branches and deliberately placed in the channel will have older versions than packages in recent Devel:Cloud:$n branches.  This is because as you know, we (unfortunately) often have PR branches which stay open for a long time, and our `_service` files use `%ct` in the `versionformat` parameter, which means that packages are versioned by the commit date on the branch HEAD, not by the package build date. 

If there is outdated crap then it should simply be removed, not used as an excuse to hinder genuine use cases.  Don't forget that `mkcloud` is for developers, not customers, so there is no argument to try to protect customers who accidentally left crap in their PTF channel.

**Comment by aspiers** [comment]
@dirkmueller please can you respond to my last comment?

**Comment by dirkmueller** [comment]
@aspiers sorry, didn't see your reply. fine with me then (I see the reason for the use case and have no suggestion on how to do it differently)

**Comment by mjura** [comment]
> If there is outdated crap then it should simply be removed, not used as an excuse to hinder genuine > use cases. Don't forget that mkcloud is for developers, not customers, so there is no argument to try > to protect customers who accidentally left crap in their PTF channel.

You right @aspiers , for me this change is OK :+1:

**Comment by rsalevsky** [comment]
+1

**Comment by bmwiedemann** [comment]
Note that we also deploy our production clouds with the qa_crowbarsetup script, so if the change is not good enough to always use, we should wrap it with
```bash
 if [[ $want_ptf_override ]] ; then
```
or such.

**Comment by aspiers** [comment]
@bmwiedemann The change should be good enough to always use.  If you don't want a PTF package applied to your next production cloud, simply remove it from the PTF repo (which sounds like a good anyway).

Even if for some reason you don't like this, qa_crowbarsetup.sh is used to deploy production clouds *far* less often than it is used to deploy test clouds, right?  i.e. multiple times per day via Jenkins and via manual developer runs, vs. once or twice every few months?  So the default behaviour should cater for deploying test clouds not production clouds.

**Pull request merged**